osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -112,7 +111,17 @@ public void Import(params string[] paths)
                     }
                 }
 
-                Import(new[] { beatmapSet });
+                try
+                {
+                    Import(beatmapSet);
+                }
+                catch
+                {
+                }
+                finally
+                {
+                    File.Delete(p);

```

Comment:
> what if the file is in use? the file.delete will cause an unhandled exception?

Yes, it will.

> the web browser downloading the file

Partial downloads usually have different extension. If we will try to import only .osz files here, this case shouldn't be a problem.